evm/p2p/lightchain.py

```diff
@@ -438,15 +442,20 @@ def peer_block_height(peer: LESPeer):
         return reply['receipts'][0]
 
     @alru_cache(maxsize=1024, cache_exceptions=False)
```

Comment:
Do you know if `alru_cache` appropriately handles decorating methods.  I know that other such decorators sometimes have trouble with the `self` argument since it normally shouldn't be included in the cache.

Reply:
It does include `self` in the cache's key, but that is the behaviour I'd expect in most cases, I think. Can you think of a situation where we'd want to reuse the cache across instances?

Anyway, I think it may not matter here as the LightChain instance is meant to be a singleton and trying to have multiple instances of it is not supported

Reply:
Sounds good to me.  I just wanted to make sure that we weren't going to lose all of the cache benefits for some reason because of this.  👍